manifests/server/access.pp

```diff
@@ -1,23 +1,27 @@
 # See README.md for details.
 define openldap::server::access (
-  Optional[Enum['present', 'absent']]  $ensure   = undef,
-  Optional[Variant[Integer,String[1]]] $position = undef, # FIXME We should probably choose Integer or String
-  Optional[String[1]]                  $what     = undef,
-  Optional[String[1]]                  $suffix   = undef,
-  Optional[Array[String[1]]]           $access   = undef,
+  String[1]                            $what,
+  Array[Openldap::Access_rule]         $access,
+  Enum['present', 'absent']            $ensure   = 'present',
+  Optional[Variant[Integer,String[1]]] $position = undef, # FIXME deprecated
+  Optional[String[1]]                  $suffix   = undef, # FIXME deprecated
 ) {
   include openldap::server
 
+  if $position or $suffix {
+    warning('openldap::server::access position and suffix are deprecated')
```

Comment:
should we also add a notify here? warnings are only logged to the server, not on the agent side.

Reply:
Yes, a notify would be better suited.

Now I am wondering if we should better just drop this since that mean for sure 2 breaking releases for access fixes, which have already introduced major releases in the past?

Reply:
IMHO : `Array[Hash[String[1], Any]]` == :+1:

Reply:
Thanks, I updated the PR accordingly.